app/models/miq_widget.rb

```diff
@@ -532,6 +537,23 @@ def sync_schedule(schedule_info)
     sched
   end
 
+  def existing_scheduler
+    return nil if (sched = MiqSchedule.find_by(:name => description)).nil?
+
+    # return existing sheduler if filter referr to the same widget
+    return sched if sched.filter == filter_for_schedule
+
+    # change name of existed schedule in case it is in use
+
+    suff = Time.new.utc.to_s
+    _log.warn("Schedule #{sched.name} already exists, renaming it to `#{sched.name} #{suff}`")
+    sched.name = "#{sched.name} #{suff}"
```

Comment:
Should we `sched.update(:name => "#{sched.name} #{suff}", :description => "#{sched.description} #{suff}")` instead?

Reply:
OK